types/coin.go

```diff
@@ -32,7 +32,7 @@ func NewCoin(denom string, amount Int) Coin {
 	}
 
 	return Coin{
-		Denom:  denom,
+		Denom:  strings.ToUpper(denom),
```

Comment:
Can we have them to be lowercase instead ?

Reply:
I'm also a fan of that personally

Reply:
Done